controllers/controller_ref_manager.go

```diff
@@ -529,3 +531,218 @@ func CheckStatefulSet(etcd *druidv1alpha1.Etcd, statefulSet *appsv1.StatefulSet)
+
+	if etcd.Spec.Backup.GarbageCollectionPolicy != nil {
+		backupValues["garbageCollectionPolicy"] = etcd.Spec.Backup.GarbageCollectionPolicy
+	}
+
+	if etcd.Spec.Backup.GarbageCollectionPeriod != nil {
+		backupValues["garbageCollectionPeriod"] = etcd.Spec.Backup.GarbageCollectionPeriod
+	}
+
+	if etcd.Spec.Backup.DeltaSnapshotPeriod != nil {
+		backupValues["deltaSnapshotPeriod"] = etcd.Spec.Backup.DeltaSnapshotPeriod
+	}
+
+	if etcd.Spec.Backup.BackupCompactionSchedule != nil {
+		backupValues["backupCompactionSchedule"] = etcd.Spec.Backup.BackupCompactionSchedule
+	}
+
+	if etcd.Spec.Backup.EtcdSnapshotTimeout != nil {
+		backupValues["etcdSnapshotTimeout"] = etcd.Spec.Backup.EtcdSnapshotTimeout
+	}
+
+	if etcd.Spec.Backup.Port != nil {
+		backupValues["port"] = etcd.Spec.Backup.Port
+	}
+
+	if etcd.Spec.Backup.SnapshotCompression != nil {
+		compressionValues := make(map[string]interface{})
+		if etcd.Spec.Backup.SnapshotCompression.Enabled {
+			compressionValues["enabled"] = etcd.Spec.Backup.SnapshotCompression.Enabled
+		}
+		if etcd.Spec.Backup.SnapshotCompression.Policy != nil {
+			compressionValues["policy"] = etcd.Spec.Backup.SnapshotCompression.Policy
+		}
+		backupValues["compression"] = compressionValues
+	}
+
+	if etcd.Spec.Backup.Image == nil {
+		val, ok := images[common.BackupRestore]
+		if !ok {
+			return map[string]interface{}{}, fmt.Errorf("either etcd resource or image vector should have %s image", common.BackupRestore)
+		}
+		backupValues["image"] = val.String()
+	} else {
+		backupValues["image"] = etcd.Spec.Backup.Image
+	}
+
+	volumeClaimTemplateName := etcd.Name
+	if etcd.Spec.VolumeClaimTemplate != nil && len(*etcd.Spec.VolumeClaimTemplate) != 0 {
+		volumeClaimTemplateName = *etcd.Spec.VolumeClaimTemplate
+	}
+
+	sharedConfigValues := map[string]interface{}{
+		"autoCompactionMode":      druidv1alpha1.Periodic,
+		"autoCompactionRetention": DefaultAutoCompactionRetention,
+	}
+
+	if etcd.Spec.Common.AutoCompactionMode != nil {
+		sharedConfigValues["autoCompactionMode"] = etcd.Spec.Common.AutoCompactionMode
+	}
+
+	if etcd.Spec.Common.AutoCompactionRetention != nil {
+		sharedConfigValues["autoCompactionRetention"] = etcd.Spec.Common.AutoCompactionRetention
+	}
+
+	values := map[string]interface{}{
+		"name":                    etcd.Name,
+		"uid":                     etcd.UID,
+		"selector":                etcd.Spec.Selector,
+		"labels":                  etcd.Spec.Labels,
+		"annotations":             etcd.Spec.Annotations,
+		"etcd":                    etcdValues,
+		"backup":                  backupValues,
+		"sharedConfig":            sharedConfigValues,
+		"replicas":                etcd.Spec.Replicas,
+		"statefulsetReplicas":     statefulsetReplicas,
+		"serviceName":             fmt.Sprintf("%s-client", etcd.Name),
+		"configMapName":           fmt.Sprintf("etcd-bootstrap-%s", string(etcd.UID[:6])),
+		"jobName":                 getJobName(etcd),
+		"volumeClaimTemplateName": volumeClaimTemplateName,
+	}
```

Comment:
Just an FYI
will have to append `roleName`, `roleBindingName`, and `serviceAccountName` here after [233](https://github.com/gardener/etcd-druid/pull/233) is merged